lib/Conversion/TorchToLinalg/Reduction.cpp

```diff
@@ -301,9 +304,73 @@ class ConvertReductionOp : public ConversionPattern {
     if (auto sumOp = dyn_cast<AtenSumDimIntListOp>(op))
       return computeReductionOpInfoFromDimOp(sumOp, operands, rewriter, opInfo);
 
+    if (auto normOp = dyn_cast<AtenLinalgVectorNormOp>(op))
+      return computeReductionOpInfoFromDimOp(normOp, operands, rewriter,
+                                             opInfo);
+
     return rewriter.notifyMatchFailure(op, "not a supported reduce op");
   }
 
+  Value payloadForNormSummation(OpBuilder &b, Location loc,
+                                ValueRange payloadArgs, Value ord,
+                                Type elemType) const {
+    // TODO: Short-circuit operations if `ord` is zero or one.
+    auto elem = payloadArgs[0];
+    auto result = payloadArgs[1];
+    auto self = convertScalarToDtype(b, loc, elem, elemType);
+    auto abs = b.create<math::AbsOp>(loc, self);
+    auto pow = b.create<math::PowFOp>(loc, abs, ord);
+    return b.create<arith::AddFOp>(loc, pow, result);
+  }
+
+  /// Generate a linalg.generic operation for performing a sum reduction along
+  /// the tensor and dimensions specified in `opInfo`, such that the element
+  /// type of the result tensor is `elemType`.
+  Value createNormSumReduction(Location loc, Type elemType,
+                               const ReductionOpInfo &opInfo, Value ord,
+                               ConversionPatternRewriter &rewriter) const {
+    auto err = false;
+
+    // Function to create the body of the linalg.generic operation.
+    auto sumBodyBuilder = [&](OpBuilder &builder, Location loc,
+                              ValueRange payloadArgs) {
+      auto result =
+          payloadForNormSummation(builder, loc, payloadArgs, ord, elemType);
+      if (result)
+        builder.create<linalg::YieldOp>(loc, result);
+      err = !result;
+    };
+
+    auto zeroAttr = rewriter.getZeroAttr(elemType);
+    auto initElement = rewriter.create<arith::ConstantOp>(loc, zeroAttr);
+
+    // linalg.generic operation for sum reduction
+    auto sumOp = torch_to_linalg::createReductionLinalgGeneric(
+        rewriter, loc, opInfo.tensorOperand, opInfo.dimSet, opInfo.keepDim,
+        initElement, sumBodyBuilder);
+    return err || !sumOp ? Value{} : sumOp;
+  }
+
+  /// Generate a linalg.generic operation for pointwise exponentiation of each
+  /// element.
+  Value createNormExp(Location loc, Type elemType, Value exponent, Value sumOp,
```

Comment:
nit: I think its better to call this `createElementwiseExp`, since the function does not require to be used for norm purposes. I would also call the argument `sumOp` `inputTensor` instead

Reply:
Fixed in commit 2d9cadd.